src/Microsoft.OData.Core/ODataInnerError.cs

```diff
@@ -30,17 +32,44 @@ public ODataInnerError()
         /// <summary>Initializes a new instance of the <see cref="T:Microsoft.OData.ODataInnerError" /> class with exception object.</summary>
         /// <param name="exception">The <see cref="T:System.Exception" /> used to create the inner error.</param>
         public ODataInnerError(Exception exception)
+            : this(exception, "internalexception")
         {
-            ExceptionUtils.CheckArgumentNotNull(exception, "exception");
 
-            this.Message = exception.Message ?? string.Empty;
-            this.TypeName = exception.GetType().FullName;
-            this.StackTrace = exception.StackTrace;
+        }
+
+        public ODataInnerError(Exception exception, string nestedObjectName)
+        {
+            ExceptionUtils.CheckArgumentNotNull(exception, "exception");
 
             if (exception.InnerException != null)
             {
-                this.InnerError = new ODataInnerError(exception.InnerException);
+                this.InnerError = new ODataInnerError(exception.InnerException, nestedObjectName);
             }
+
+            Properties = new Dictionary<string, string>();
+            Properties.Add("message", exception.Message ?? string.Empty);
+            Properties.Add("type", exception.GetType().FullName ?? string.Empty);
+            Properties.Add("stacktrace", exception.Message ?? string.Empty);
+        }

```

Comment:
So the Message, TypeName, and Stacktrace properties are no longer used?  If I have existing code that creates an inner error and then sets these, those values will no longer be used?   And if I create an instance I can no longer get those values from the properties?